src/CommonLib/utility.cxx

```diff
@@ -606,4 +567,34 @@ UTILITY::FindDotNetFolders(
     } while (FindNextFileW(handle, &data));
 
     FindClose(handle);
-}
\ No newline at end of file
+}
+
+BOOL
+UTILITY::CheckIfFileExists(
+    _In_ PCWSTR pszFilePath
+)
+{
+    HANDLE              hFileHandle = INVALID_HANDLE_VALUE;
+    SECURITY_ATTRIBUTES saAttr;
+    BOOL                fFileExists;
+
+    saAttr.nLength = sizeof(SECURITY_ATTRIBUTES);
+    saAttr.bInheritHandle = TRUE;
+    saAttr.lpSecurityDescriptor = NULL;
+
+    hFileHandle = CreateFile(pszFilePath,
```

Comment:
if the file is opened by another thread for writing, CreateFile will fail. You may need to check the last error code to see whether it is permission issue

Reply:
Any reason why I can't use PathFileExists? https://msdn.microsoft.com/en-us/library/windows/desktop/bb773584(v=vs.85).aspx

Reply:
you can. the limitation is long file path support (which we claimed on Win10) and dependency on Shlwapi.dll.

Reply:
We already have that dependency on that dll. Considering we are avoiding file path issues already, I will continue to use CreateFile.

Reply:
then we can use it.